src/Application.php

```diff
@@ -132,17 +132,23 @@ public function __call($method, $args)
         if (! in_array(strtoupper($method), $this->httpRouteMethods, true)) {
             throw new BadMethodCallException('Unsupported method');
         }
-
-        if (count($args) !== 2) {
+        $numArgs = count($args);
+        if ($numArgs < 2) {
             throw new BadMethodCallException(sprintf(
-                '%s::%s requires exactly 2 arguments; received %d',
+                '%s::%s requires at least 2 arguments; received %d',
                 __CLASS__,
                 $method,
                 count($args)
             ));
         }
 
-        $args[] = [$method];
+        if ($numArgs === 3) {
+            $name    = $args[2];
+            $args[2] = [$method];
+            $args[]  = $name;
```

Comment:
I don't think we should support passing the name as part of the argument list:

- We already default to using the route match string if no name is provided.
- Because `setName()` is available on the returned `Route` object, it can be set after-the-fact.
- If we support it, next, developers will want `$options` as a potential argument, and then we have issues with argument order to contend with.

Since name and options are both optional and can be set on the `Route` instance, let's keep them out of the signature.